apps/sms/views/conversation/js/conversation.js

```diff
@@ -2957,52 +2842,42 @@ var ConversationView = {
     new OptionMenu(params).show();
   },
 
-  discardDraft: function conv_discardDraft() {
-    // If we were tracking a draft properly, update the Drafts object entry.
-    if (!this.draft) {
-      return;
-    }
+  updateDraft() {
+    var thread = Threads.active;
 
-    Drafts.delete(this.draft).store();
+    // Need to assimilate recipients in order to check if any entered.
+    this.assimilateRecipients();
 
-    this.draft = null;
-  },
+    // If message content is not empty and/or we have _valid_ recipients in
+    // NewMessage view.
+    var hasDraftToSave = !Compose.isEmpty() ||
+      (!thread && this.recipients.numbers.length);
```

Comment:
Yep, at first I thought that it would be as easy as just replacing ```this.recipients.numbers.length``` with ```this.recipients.length``` - but it just partially worked - so yeah it's better to handle this separately in https://github.com/mozilla-b2g/gaia/pull/29498